.github/workflows/build-and-test.yaml

```diff
@@ -22,15 +22,12 @@ jobs:
             build-depends-repos: build_depends.repos
     steps:
       - name: Check out repository
-        uses: actions/checkout@v3
-
-      - name: Free disk space (Ubuntu)
-        uses: jlumbroso/free-disk-space@v1.3.1
```

Comment:
## Reason for removing free-disk-space action

This action clears space if ran directly under the github hosted machine.

But we are running this workflow in a container. Thus this action doesn't do anything.